core/src/validator.rs

```diff
@@ -194,13 +194,12 @@ impl Validator {
                         accounts
                             .iter()
                             .filter_map(|(pubkey, (stake, account))| {
-                                if let Ok(vote_state) = VoteState::deserialize(&account.data) {
-                                    if *stake == 0 {
-                                        warn!("Validator with no stake exiting on first ^C");
+                                if *pubkey == keypair.pubkey() && *stake > 0 {
+                                    if let Ok(vote_state) = VoteState::deserialize(&account.data) {
+                                        warn!("Validator with stake is exiting on first ^C");
                                         std::process::exit(1);
```

Comment:
@mvines , do u think i should return Some(*pubkey, *stake) somewhere here?

Reply:
I'd just return `Some(*stake)` here.   Then turn `epoch_vote_accounts(..).map(|accounts| ...)` into `epoch_vote_accounts(..).filter_map(|accounts| ...)`.  This way once we arrive at line 206 we have either
1. value `None` if the node is not stake, in which can we can `std::process::exit(1)`
2. `Some(stake)` representing the current stake of the node, in which can we can `warn!` the user